### PR TITLE
Document interfaces

### PR DESCRIPTION
**Pull request by caiotomazelli**

### CODE REVIEW

File: tightlock_api/app/main.py [hunk 1]
```diff
@@ -176,6 +250,20 @@ async def batch_get_activations_runs(
...
+  
+  Args:
+    session: Postgres DB session dependency injection.
+    airflow_client: Airflow Client dependency injection.
+    activation_names: List of interested activations to retrieve logs from.
+      Defaults to None, when logs from all activations are retrieved.
+    page: Page number to be used by paginating clients. 
+      Defaults to zero, when no offset is applied.
+    page_size: Size of page to be used by paginating clients.
+      Default to 20.
```
Comment (HalleyInteractive) [inline tightlock_api/app/main.py:263]: `Defaults to 20`

**Review by HalleyInteractive — CHANGES_REQUESTED** [review]

File: dags/protocols/destination_proto.py [hunk 2]
```diff
@@ -11,43 +11,78 @@
...
 
 @runtime_checkable
 class DestinationProto(Protocol):
+  """Common set of methods that must be implemented by all destinations."""
 
   def __init__(self, config: Dict[str, Any]):
+    """Init method for DestinationProto.
+    
+    Args:
+      config: A dictionnary parsed from the connection config, containing
```
[the remark this answers is not in the document]
Comment (caiotomazelli) [reply dags/protocols/destination_proto.py:31]: Done

File: dags/protocols/source_proto.py [hunk 3]
```diff
@@ -33,10 +40,35 @@ def get_data(
...
+      A list of field-value mappings retrieved from the target data source.
+    """
     ...
 
   def schema(self) -> Dict[str, Any]:
+    """Returns the required metadata for this source config.
+    
+    Returns:
+      A config dict, preferably generated by defining a schema class
+      that inherets from Pydantic BaseModel and calling the
```
[the remark this answers is not in the document]
Comment (caiotomazelli) [reply dags/protocols/source_proto.py:62]: Done

File: tightlock_api/app/main.py (hunk 1 above)
Comment (caiotomazelli) [reply tightlock_api/app/main.py:263]: Done

File: dags/protocols/destination_proto.py [hunk 4]
```diff
@@ -11,43 +11,78 @@
...
+    """
     ...
 
-  def batch_size(self) -> int:
+  def schema(self) -> Dict[str, Any]:
+    """Returns the required metadata for this destination config.
+    
+    Returns:
+      A config dict, preferably generated by defining a schema class
+      that inherets from Pydantic BaseModel and calling the
```
[the remark this answers is not in the document]
Comment (caiotomazelli) [reply dags/protocols/destination_proto.py]: Done

**Review by HalleyInteractive — APPROVED** [review]

**Pull request merged**